lib/screens/wallet/components/tokens_cards/interactor/mappers/token_balances_state_mapper.dart

```diff
@@ -37,7 +37,7 @@ class TokenBalancesStateMapper {
           }
         } else {
           final BalanceModel balance = result.asValue!.value;
-          if (whitelisted || balance.quantity > 0) {
+          if (whitelisted || balance.quantity != 0) {
```

Comment:
So we now have negative balances?

The reason I never integrated the token master contract methods is that this way of showing tokens does not scale

Instead we need to do what we did in the Hypha wallet, users see default tokens, and a + button, and can add tokens. 

When they add tokens, we load the entire list of available tokens, each with a toggle 

We store which tokens they show

Seeds Light Wallet tried to be overly smart and just show all tokens that have a balance - but it's too many network requests, and I think it may even be blocking the UX. So we can't do that (my fault... )

Reply:
Totally agree something like this is essential and Hypha Wallet seems to do it well. Also we should distinguish between badge-like and currency-like tokens in the UI.

Doesn't have to be in this PR though.

Reply:
See this ticket https://github.com/JoinSEEDS/seeds_light_wallet/issues/1951